src/pymortests/vectorarray.py

```diff
@@ -153,6 +153,13 @@ def test_zeros(vector_array):
             assert np.allclose(d, np.zeros((c, v.dim)))
 
 
+def test_from_data(vector_array):
+    if hasattr(vector_array, 'data'):
+        d = vector_array.data
+        v = vector_array.from_data(d, vector_array.subtype)
```

Comment:
One should catch `NotImplementedError` here. Not every `VectorArray` which allows read access to the data may allow write access, so implementing `data` is not enough to guarantee that this line will work.